### PR TITLE
[Test Coverage] Refactor readline input validation for testability

### LINKED ISSUE #322: [Test Coverage] Refactor readline input validation for testability
## Summary
Extract input validation logic from readline functions to make them testable and increase coverage from 0% to ~18% for `cli_readline.go`.

## Parent Issue
Part of #248 - Improve test coverage for readline and system variables

## Current Status
- **File**: `cli_readline.go` (303 lines)
- **Current coverage**: Almost 0%
- **Target functions**:
  - `initializeMultilineEditor` (line 88): 0.0% → 60%
  - `readInteractiveInput` (line 269): 0.0% → 70%
  - `PS1PS2FuncToPromptFunc` (line 132): 0.0% → 100%

### 1. Extract Pure Functions
Create testable functions for:
- Statement completion detection logic
- Waiting status management
- Delimiter detection and validation
- Prompt generation logic

### 2. Refactor Core Functions
- `initializeMultilineEditor`:
  - Extract `SubmitOnEnterWhen` logic into pure function
  - Separate prompt configuration from editor setup
- `readInteractiveInput`:
  - Extract input validation logic
  - Separate error handling from input processing

### 3. Create Comprehensive Tests
Test scenarios:
- Complete vs incomplete statements
- Various delimiter types
- Error status handling
- Multi-line input processing
- Edge cases (empty input, special characters)

## Technical Approach
```go
// Example: Extract statement validation
func shouldSubmitStatement(statements []inputStatement, err error, waitingStatus string) (bool, string) {
    // Pure logic for determining if statement should be submitted
}

// Example: Extract prompt logic
func generatePrompt(systemVars SystemVariables, lineNum int) string {
    // Pure prompt generation
}
```

## Expected Coverage Gain
- **Lines to cover**: ~56 lines
- **Impact on cli_readline.go**: 0% → ~18%
- **New testable functions**: 4-6 pure functions

## Success Criteria
- [ ] Input validation logic extracted to pure functions
- [ ] All extracted functions have >90% test coverage
- [ ] Original behavior preserved
- [ ] Tests cover all statement types and edge cases
- [ ] Improved code maintainability through separation of concerns

### PR DESCRIPTION
**Pull request by apstndb**
## Summary
This PR refactors readline input validation logic to improve testability and increases test coverage for `cli_readline.go` as part of issue #322.

### Extracted Pure Functions
Created testable pure functions by extracting logic from existing code:

1. **`shouldSubmitStatement`** - Determines when to submit input based on statements and errors
   - Extracted from `SubmitOnEnterWhen` callback in `initializeMultilineEditor`
   - Handles lexer status errors and statement completion detection

2. **`processInputLines`** - Converts readline output to inputStatement with error handling
   - Extracted from `readInteractiveInput`
   - Handles read errors while preserving partial input

3. **`validateInteractiveInput`** - Validates single statement constraint for interactive mode
   - Extracted from `readInteractiveInput`
   - Ensures only one statement is processed in interactive mode

4. **`generatePS2Prompt`** - Generates continuation prompts with optional padding
   - Extracted from prompt configuration in `initializeMultilineEditor`
   - Supports `%P` prefix for automatic padding alignment

5. **`PS1PS2FuncToPromptFunc`** - Converts PS1/PS2 functions to readline prompt function
   - Already existed but previously untested
   - Now has comprehensive test coverage

### Test Coverage Improvements
- Added comprehensive tests for all 5 functions
- Achieved **100% coverage** for all extracted functions
- Tests cover:
  - Statement completion detection
  - Error handling (lexer errors, read errors)
  - Multi-line input processing
  - Prompt formatting and padding
  - Edge cases (empty input, multiple statements)

### Coverage Results
- 5 functions now have 100% test coverage
- Total of 37 lines covered out of 82 tracked lines in coverage data
- Significant improvement in readline validation testability

## Testing
- All tests pass: `make check` ✅
- No linting issues
- Existing behavior preserved

Fixes #322

EOF < /dev/null

### CODE REVIEW

File: cli_readline.go [hunk 1]
```diff
@@ -100,30 +119,18 @@ func initializeMultilineEditor(c *Cli) (*multiline.Editor, History, error) {
...
 			return c.getInterpolatedPrompt(c.SystemVariables.Prompt)
 		},
 		func(ps1 string) string {
-			lastLineOfPrompt := lo.LastOrEmpty(strings.Split(ps1, "\n"))
-
-			prompt2, needPadding := strings.CutPrefix(c.SystemVariables.Prompt2, "%P")
-			interpolatedPrompt2 := c.getInterpolatedPrompt(prompt2)
-			return lo.Ternary(needPadding, runewidth.FillLeft(interpolatedPrompt2, runewidth.StringWidth(lastLineOfPrompt)), interpolatedPrompt2)
+			interpolatedPrompt2 := c.getInterpolatedPrompt(c.SystemVariables.Prompt2)
+			return generatePS2Prompt(ps1, c.SystemVariables.Prompt2, interpolatedPrompt2)
```
[the remark this answers is not in the document]
Comment (apstndb) [reply cli_readline.go]: Thank you for the feedback!

Fixed in commit 1fdf6d0.

File: cli_readline.go [hunk 2]
```diff
@@ -272,28 +280,25 @@ func setupHistory(ed *multiline.Editor, historyFileName string) (History, error)
...
+	if readErr != nil {
 		if len(lines) == 0 {
-			return nil, fmt.Errorf("failed to read input: %w", err)
+			return nil, fmt.Errorf("failed to read input: %w", readErr)
 		}
 
 		str := strings.Join(lines, "\n")
 		return &inputStatement{
 			statement:                str,
 			statementWithoutComments: str,
```
[the remark this answers is not in the document]
Comment (apstndb) [reply cli_readline.go:293]: This behavior preserves the original implementation. When readline returns an error with partial input, both `statement` and `statementWithoutComments` are intentionally set to the same raw string. This is the expected behavior for error recovery paths - comment stripping is only performed during normal input processing via `separateInput`. The test cases already cover this scenario in 'read error with non-empty lines' and 'single line with error'.
